spec/state_machines/form_state_machine_spec.rb

```diff
@@ -69,17 +69,19 @@ class FakeForm < Form
   end
 
   describe ".archive_live_form" do
-    let(:form) { FakeForm.new(state: :live) }
+    context "when the form is live" do
+      let(:form) { FakeForm.new(state: :live) }
 
-    it "transitions to draft if form is live" do
-      expect(form).to transition_from(:live).to(:draft).on_event(:archive_live_form)
+      it "transitions to archived" do
+        expect(form).to transition_from(:live).to(:archived).on_event(:archive_live_form)
+      end
     end
 
     context "when form is live_with_draft" do
```

Comment:
I think we should add an unhappy path here of a draft form calling this event. It should raise an error saying it cant transition to state

Reply:
I've added it, I'm a bit wary of doing too many unhappy state change paths as there can be so many!

Reply:
https://github.com/alphagov/forms-api/pull/439/files#diff-e80deed9471f0979ccc4d5b60fe3e503adcafd7466d9e312ebffa7ad8bf093fbR72-R78